examples/erc1155/lib.rs

```diff
@@ -0,0 +1,827 @@
+        #[ink(message)]
+        fn safe_transfer_from(
+            &mut self,
+            from: AccountId,
+            to: AccountId,
+            token_id: TokenId,
+            value: Balance,
+            data: Vec<u8>,
+        ) -> Result<()> {
+            let caller = self.env().caller();
+            if caller != from {
+                ensure!(self.is_approved_for_all(from, caller), Error::NotApproved);
+            }
+
+            ensure!(to != AccountId::default(), Error::ZeroAddressTransfer);
+
+            let balance = self.balance_of(from, token_id);
+            ensure!(balance >= value, Error::InsufficientBalance);
+
+            self.perform_transfer(from, to, token_id, value);
+            self.transfer_acceptance_check(caller, from, to, token_id, value, data);
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        fn safe_batch_transfer_from(
+            &mut self,
+            from: AccountId,
+            to: AccountId,
+            token_ids: Vec<TokenId>,
+            values: Vec<Balance>,
+            data: Vec<u8>,
+        ) -> Result<()> {
+            let caller = self.env().caller();
+            if caller != from {
+                ensure!(self.is_approved_for_all(from, caller), Error::NotApproved);
+            }
+
+            ensure!(to != AccountId::default(), Error::ZeroAddressTransfer);
+            ensure!(!token_ids.is_empty(), Error::BatchTransferMismatch);
+            ensure!(
+                token_ids.len() == values.len(),
+                Error::BatchTransferMismatch,
+            );
+
+            let transfers: Vec<(_, _)> = token_ids.iter().zip(values.iter()).collect();
+            for (&id, &v) in &transfers {
+                let balance = self.balance_of(from, id);
+                ensure!(balance >= v, Error::InsufficientBalance);
+            }
+
+            for (&id, &v) in &transfers {
+                self.perform_transfer(from, to, id, v);
+            }
+
+            // Can use the any token ID/value here, we really just care about knowing if `to` is a
+            // smart contract which accepts transfers
+            self.transfer_acceptance_check(
+                caller,
+                from,
+                to,
+                token_ids[0],
+                values[0],
+                data,
+            );
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        fn balance_of(&self, owner: AccountId, token_id: TokenId) -> Balance {
+            *self.balances.get(&(owner, token_id)).unwrap_or(&0)
+        }
+
+        #[ink(message)]
+        fn balance_of_batch(
+            &self,
+            owners: Vec<AccountId>,
+            token_ids: Vec<TokenId>,
+        ) -> Vec<Balance> {
+            let mut output = Vec::new();
+            for o in &owners {
+                for t in &token_ids {
+                    let amt = self.balance_of(*o, *t);
+                    output.push(amt);
+                }
+            }
```

Comment:
```suggestion
            for o in owners {
                for t in token_ids {
                    let amount = self.balance_of(o, t);
                    output.push(amount);
                }
            }
```

Reply:
Nevermind, the compiler wasn't happy about the whole nested `into_iter()` thing